### PR TITLE
chore: delegate variant pushes to the right method

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It groups the conditions for pushing the docker image of each beat, because the logic for the variants is responsible of pushing or logging a warning if the image exists.

## Why is it important?
It won't push some of the images twice, also simplifying the IF/ELSE logic.

Apart from that, there are wrong image names after the addition of variants, like `filebeat-oss-oss`.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [x] Groovy file validated

## Related issues

- Caused by #21621

### CODE REVIEW

**Review by cachedout — APPROVED** [review]

**Comment by mdelapenya** [comment]
The pushes for this PR are OK. See https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackaging/detail/PR-21861/2/pipeline/185

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -173,30 +173,20 @@ pipeline {
...
-    } else if ("${env.BEATS_FOLDER}" == "filebeat") {
-      tagAndPush('filebeat-oss')
-    } else if ("${env.BEATS_FOLDER}" == "heartbeat"){
-      tagAndPush('heartbeat-oss')
+    if ("${env.BEATS_FOLDER}" == "auditbeat" || "${env.BEATS_FOLDER}" == "x-pack/auditbeat"){
+      tagAndPush('auditbeat')
+    } else if ("${env.BEATS_FOLDER}" == "filebeat" || "${env.BEATS_FOLDER}" == "x-pack/filebeat") {
+      tagAndPush('filebeat')
+    } else if ("${env.BEATS_FOLDER}" == "heartbeat" || "${env.BEATS_FOLDER}" == "x-pack/heartbeat"){
+      tagAndPush('heartbeat')
```
Comment (v1v) [inline .ci/packaging.groovy:181]: NIT: using endsWith and `?` to access the elements from the `env` dict

```suggestion
    if (env?.BEATS_FOLDER?.endsWith('auditbeat')) {
      tagAndPush('auditbeat')
    } else if (env?.BEATS_FOLDER?.endsWith('filebeat')) {
      tagAndPush('filebeat')
    } else if (env?.BEATS_FOLDER?.endsWith('heartbeat')) {
      tagAndPush('heartbeat')
```

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -173,30 +173,20 @@ pipeline {
...
     } else if ("${env.BEATS_FOLDER}" == "journalbeat"){
       tagAndPush('journalbeat')
-      tagAndPush('journalbeat-oss')
-    } else if ("${env.BEATS_FOLDER}" == "metricbeat"){
-      tagAndPush('metricbeat-oss')
+    } else if ("${env.BEATS_FOLDER}" == "metricbeat" || "${env.BEATS_FOLDER}" == "x-pack/metricbeat"){
+      tagAndPush('metricbeat')
```
Comment (v1v) [inline .ci/packaging.groovy:185]:
```suggestion
    } else if (env?.BEATS_FOLDER?.endsWith('metricbeat')) {
      tagAndPush('metricbeat')
```

**Review by v1v — APPROVED** [review]
👍

**Pull request merged**